### PR TITLE
Browser test all form editor fields

### PR DESCRIPTION
**Pull request by aatkin**
relates #2189 

* refactor existing `test-form-editor` browser test with utility functions
* split field creation in `test-form-editor` with `testing` macro (a bit easier to read separate phases)
* add following fields to `test-form-editor`:
  * multi-select
  * table
  * email
  * phone number
  * ip address
  * attachment
  * label
  * header

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue

## Testing
- [x] Valuable features are integration / browser / acceptance tested automatically

## Follow-up
- [ ] New tasks are created for pending or remaining tasks

### CODE REVIEW

File: test/clj/rems/test_browser.clj [hunk 1]
```diff
@@ -1354,65 +1386,280 @@
...
+            (is (btu/eventually-visible? [{:id (str "container-form-1-field-fld" idx)}
+                                          {:tag :button :fn/has-class "info-button"}]))
+
+            (btu/scroll-and-click [{:id (str "container-form-1-field-fld" idx)}
+                                   {:tag :button :fn/has-class "info-button"}])
+            (is (btu/eventually-visible? {:id (str "form-1-field-fld" idx "-collapse")}))
+            (is (= (btu/value-of {:id (str "form-1-field-fld" idx "-collapse")})
+                   "Info text (EN)"))))
+
+        (testing "create email address field"
```
Comment (aatkin) [inline test/clj/rems/test_browser.clj:1584]: `email`, `phone` and `ip address` fields are similar text input. i think these should be tested by triggering validation but that does not show up in the preview currently

Comment (aatkin) [reply test/clj/rems/test_browser.clj:1594]: as discussed, validation currently not implemented in preview and thus can be disregarded so far

Comment (Macroz) [reply test/clj/rems/test_browser.clj:1594]: We have tests that check the types in the application so I think that's enough.

File: test/clj/rems/test_browser.clj [hunk 2]
```diff
@@ -1354,65 +1400,341 @@
...
+          (create-context-field! :create-form-field/email)
+
+          (let [id (field-id :create-form-field/email)]
+            (btu/scroll-and-click-el (last (btu/query-all {:class :add-form-field})))
+            (btu/scroll-and-click (field-selector :type-email))
+            (is (btu/eventually-visible? {:css (str "input"
+                                                    "#form-1-field-" id
+                                                    "[type='email']")}))
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:1602]: Same question here that could this be just an id selector?

File: test/clj/rems/test_browser.clj [hunk 3]
```diff
@@ -1354,65 +1400,341 @@
...
+
+        (testing "create phone number field"
+          (create-context-field! :create-form-field/phone-number)
+
+          (let [id (field-id :create-form-field/phone-number)]
+            (btu/scroll-and-click-el (last (btu/query-all {:class :add-form-field})))
+            (btu/scroll-and-click (field-selector :type-phone-number))
+            (is (btu/eventually-visible? {:css (str "input"
+                                                    "#form-1-field-" id
+                                                    "[type='tel']")}))
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:1630]: And here.

File: test/clj/rems/test_browser.clj [hunk 4]
```diff
@@ -1354,65 +1400,341 @@
...
+
+        (testing "create ip address field"
+          (create-context-field! :create-form-field/ip-address)
+
+          (let [id (field-id :create-form-field/ip-address)]
+            (btu/scroll-and-click-el (last (btu/query-all {:class :add-form-field})))
+            (btu/scroll-and-click (field-selector :type-ip-address))
+            (is (btu/eventually-visible? {:css (str "input"
+                                                    "#form-1-field-" id
+                                                    "[type='text']")}))
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:1658]: And here.

File: test/clj/rems/test_browser.clj [hunk 5]
```diff
@@ -1354,65 +1400,341 @@
...
+
+        (testing "create attachment field"
+          (create-context-field! :create-form-field/attachment)
+
+          (let [id (field-id :create-form-field/attachment)]
+            (btu/scroll-and-click-el (last (btu/query-all {:class :add-form-field})))
+            (btu/scroll-and-click (field-selector :type-attachment))
+            (is (btu/eventually-invisible? {:css (str "input"
+                                                      "#upload-form-1-field-" id "-input"
+                                                      "[type='file']")}))
```
Comment (Macroz) [inline test/clj/rems/test_browser.clj:1685]: ...

**Review by Macroz — APPROVED** [review]
Looks good. You can check the docstring and optionally the other stuff.

File: test/clj/rems/test_browser.clj [hunk 6]
```diff
@@ -1341,7 +1341,53 @@
                 "Active" true}
                (dissoc (slurp-fields :catalogue-item) "Start")))))))
 
+(defn create-context-field!
+  "Utility function that keeps track of created form fields in
+   context object. Stores form field in stack with `id`, that can
+   optionally be passed in `opts`. Form field can also be created into
+   bottom of stack using `:insert-first` in `opts`.
```
[the remark this answers is not in the document]
Comment (aatkin) [reply test/clj/rems/test_browser.clj:1348]: most likely yes! :+1:

File: test/clj/rems/test_browser.clj [hunk 7]
```diff
@@ -1354,65 +1400,341 @@
...
+
+        (testing "create date field"
+          (create-context-field! :create-form-field/date)
+
+          (let [id (field-id :create-form-field/date)]
+            (btu/scroll-and-click-el (last (btu/query-all {:class :add-form-field})))
+            (btu/scroll-and-click (field-selector :type-date))
+            (is (btu/eventually-visible? {:css (str "input"
+                                                    "#form-1-field-" id
+                                                    "[type='date']")}))
```
[the remark this answers is not in the document]
Comment (aatkin) [reply test/clj/rems/test_browser.clj:1575]: i originally intended the specificity as simple validation on the field type. i guess it's strictly not needed as we don't trigger any field validations either.

selector still needs to be composed but it i guess it brings the complexity down a little bit so let's do that

**Pull request merged**